### PR TITLE
feat: added ability to send user to notification list page

### PR DESCRIPTION
**Pull request by rmathis**
Allows the notification-icon component to link to an internal uPortal page (defaults to notification-list) with a queryParam highlight to mark the intended notification.

### CODE REVIEW

File: @uportal/notification-icon/src/components/NotificationIcon.vue [hunk 1]
```diff
@@ -85,6 +87,10 @@ export default {
         debug: {
             type: Boolean,
             default: false
+        },
+        navigationStrategy: {
+            type: String,
+            default: 'link'
```
Comment (ChristianMurphy) [inline @uportal/notification-icon/src/components/NotificationIcon.vue]: maybe a custom validator with the types
```js
validator: value => ['link', 'list'].indexOf(value) >= 0
```

File: @uportal/notification-icon/src/components/NotificationItem.vue [hunk 2]
```diff
@@ -14,7 +14,9 @@ import DropdownItem from 'bootstrap-vue/es/components/dropdown/dropdown-item';
 export default {
     name: 'NotificationItem',
     props: {
-        notification: Object
+        notification: Object,
+        strategy: String,
```
Comment (ChristianMurphy) [inline @uportal/notification-icon/src/components/NotificationItem.vue]: maybe adding validation
```js
validator: value => ['link', 'list'].indexOf(value) >= 0
```

File: @uportal/notification-list/src/components/NotificationList.vue [hunk 3]
```diff
@@ -116,7 +117,7 @@ export default {
         this.fetchNotificationData();
 
         const queryParams = queryString.parse(location.search);
-        this.highlight = queryParams.highlight;
+        this.highlight = queryParams.highlight || queryParams.pP_highlight;
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationList.vue]: `pP_highlight` ? where is this value coming from?

Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationList.vue]: uPortal converts `highlight` to pP_highlight when it catches the path to the notification list. I was intending to ask Drew about this to see if there is a better way of handling it.

Comment (ChristianMurphy) [reply @uportal/notification-list/src/components/NotificationList.vue]: /cc @drewwills

Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationList.vue]: I've discussed this with @drewwills and he explained that is is a necessary evil atm due to uPortal's inner workings around query parameters.

File: @uportal/notification-icon/src/components/NotificationItem.vue [hunk 4]
```diff
@@ -48,7 +60,7 @@ export default {
             return () => {};
         },
         target() {
-            return this.redirectAction ? '_self' : '_blank';
+            return this.redirectAction || this.strategy === 'list' ? '_self' : '_blank';
```
Comment (drewwills) [inline @uportal/notification-icon/src/components/NotificationItem.vue]: I believe we need to add `rel="noopener noreferrer"` everywhere we use `target="_blank"`, like here:  https://github.com/Jasig/uPortal/blob/master/uPortal-webapp/src/main/resources/layout/theme/respondr/navigation.xsl#L331

**Pull request merged**